TransDiffer/MainWindow.xaml.cs

```diff
@@ -384,26 +412,66 @@ private void Hyperlink_OnClick(object sender, RoutedEventArgs e)
         private void FileContents_OnSelectionChanged(object sender, RoutedEventArgs e)
         {
             var s = FileContents.SelectedItems.Cast<FileLineItem>().SelectMany(i => i.Tag.Strings).Distinct().ToList();
+            DialogDefinition previewDlg = null;
             if (s.Count == 1)
             {
-                CurrentDetails = s.First().String.CreateDetailsDocument(NavigateToTranslation, NavigateToFile);
+                var item = s.First();
+                CurrentDetails = item.String.CreateDetailsDocument(NavigateToTranslation, NavigateToFile);
+                previewDlg = item.Entry as DialogDefinition;
+                if (previewDlg == null)
+                {
+                    var parent = item.String?.Parent;
+                    if (parent != null)
+                    {
+                        foreach (var parentRef in item.Source.NamedLines.Values)
+                        {
+                            if (parentRef.Id == parent.Name)
+                                previewDlg = parentRef.Entry as DialogDefinition;
+                        }
+                    }
+                }
             }
-            else if (FoldersTree.SelectedItem is LangFile f)
+            else if (FoldersTree.SelectedItem is LangFile)
             {
+                var f = FoldersTree.SelectedItem as LangFile;
```

Comment:
Just a nitpick. Since the if already checked the type, there's no need to use the "safe cast" here, you can just make it `var f = (LangFile)FoldersTree.SelectedItem`.